### PR TITLE
feat(Environment Picker): Simplify the environment picker UX

### PR DESCRIPTION
**Pull request by gatzjames**
Highlights:
- [x] Updates the environment picker to have a simpler UX and make it easier to understand the order environments are applied

| Before | After |
|--------|-------|
| image | image |

Closes INS-4488

### CODE REVIEW

File: packages/insomnia/src/ui/components/environment-picker.tsx [hunk 1]
```diff
@@ -34,12 +34,12 @@ export const EnvironmentPicker = ({
...
+  const collectionEnvironmentList = [...subEnvironments].map(({ type, ...environment }) => ({ ...environment, id: environment._id, isBase: environment._id === baseEnvironment._id }));
 
   const selectedGlobalBaseEnvironmentId = activeGlobalEnvironment?.parentId.startsWith('wrk') ? activeGlobalEnvironment._id : activeGlobalEnvironment?.parentId;
   const selectedGlobalBaseEnvironment = globalBaseEnvironments.find(e => e._id === selectedGlobalBaseEnvironmentId);
 
-  const globalEnvironmentList = selectedGlobalBaseEnvironment ? [selectedGlobalBaseEnvironment, ...globalSubEnvironments.filter(e => e.parentId === selectedGlobalBaseEnvironment._id)].map(({ type, ...subenvironment }) => ({ ...subenvironment, id: subenvironment._id, isBase: subenvironment._id === selectedGlobalBaseEnvironment._id })) : [];
+  const globalEnvironmentList = selectedGlobalBaseEnvironment ? [...globalSubEnvironments.filter(e => e.parentId === selectedGlobalBaseEnvironment._id)].map(({ type, ...subenvironment }) => ({ ...subenvironment, id: subenvironment._id, isBase: subenvironment._id === selectedGlobalBaseEnvironment._id })) : [];
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/environment-picker.tsx]: :exploding_head:

File: packages/insomnia/src/ui/components/environment-picker.tsx [hunk 2]
```diff
@@ -169,6 +127,7 @@ export const EnvironmentPicker = ({
                   return match;
                 }}
                 onSelectionChange={environmentId => {
+                  console.log({ environmentId });
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/environment-picker.tsx]:
```suggestion
```

File: packages/insomnia/src/ui/components/environment-picker.tsx [hunk 3]
```diff
@@ -25,15 +30,34 @@ export const EnvironmentPicker = ({
...
+    }
+  }, [organizationId, permissionsFetcher]);
+
+  const { featuresPromise } = permissionsFetcher.data || {};
+  const [features = {
+    gitSync: { enabled: false, reason: 'Insomnia API unreachable' },
+  }] = useLoaderDeferData(featuresPromise);
+
+  const isUsingInsomniaCloudSync = Boolean(isRemoteProject(activeProject) && !activeWorkspaceMeta?.gitRepositoryId);
+  const isUsingGitSync = Boolean(features.gitSync.enabled && activeWorkspaceMeta?.gitRepositoryId);
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/environment-picker.tsx]: could this cause refetches?

**Review by filfreire — DISMISSED** [review]

File: packages/insomnia/src/ui/components/environment-picker.tsx (hunk 3 above)
Comment (gatzjames) [reply packages/insomnia/src/ui/components/environment-picker.tsx]: Good instinct. Double checked with some logs and the network open and it's safe to run.
- This was copied from other places where we grab the permissions so the same mechanics apply.
- If it's already fetched it won't rerun
- The key in the fetcher makes sure we use one fetcher on the page

**Review by jackkav — APPROVED** [review]

**Pull request merged**